projects/igniteui-angular/src/lib/services/transaction/base-transaction.ts

```diff
@@ -157,6 +157,8 @@ export class IgxBaseTransactionService<T extends Transaction, S extends State> i
             state = { value: this.cloneStrategy.clone(transaction.newValue), recordRef, type: transaction.type } as S;
             states.set(transaction.id, state);
         }
+
+        this.cleanState(transaction.id, states);
```

Comment:
This seams reasonable. Since the base only supports the pending stack (for row edit) it also didn't compare against the original. Guessing that's the note about testing with `batchEdit` on and off.

I'd actually expand on that - there's also Row Adding that relies on this that should be verified.
There's also the Hierarchical variant of the service that's specialized for the Tree Grid - so that should also be tested. As far as I can see the HierarchicalTransactionService also overrides this `updateState` method, so it _could_ be affected, though it seems like it's handling a very specific additional case.